player.gd

```diff
@@ -0,0 +1,77 @@
+extends CharacterBody3D
+
+
+const DEFAULT_SPEED = 10.0
+var SPEED = DEFAULT_SPEED
+const JUMP_VELOCITY = 6
+@onready var neck := $Neck
+@onready var camera := $Neck/Camera
+@onready var win_message_label = $"/root/Main/CanvasLayer/Label"
+
+func _unhandled_input(event: InputEvent) -> void:
+	if event is InputEventMouseButton:
+		Input.mouse_mode = Input.MOUSE_MODE_CAPTURED
+	elif event.is_action_pressed("ui_cancel"):
+		Input.mouse_mode = Input.MOUSE_MODE_VISIBLE
+	if Input.mouse_mode == Input.MOUSE_MODE_CAPTURED:
+		if event is InputEventMouseMotion:
+			neck.rotate_y(-event.relative.x * 0.01)
+			camera.rotate_x(-event.relative.y * 0.01)
+			camera.rotation.x = clamp(camera.rotation.x, deg_to_rad(-60), deg_to_rad(60))
+
+func _physics_process(delta: float) -> void:
+	# Add the gravity.
+	if not is_on_floor():
+		velocity += get_gravity() * delta
+
+	# Handle jump.
+	if Input.is_action_just_pressed("ui_accept") and is_on_floor():
+		velocity.y = JUMP_VELOCITY
+	if Input.is_action_pressed("turn_left"):
+		neck.rotate_y(0.04)
+	if Input.is_action_pressed("turn_right"):
+		neck.rotate_y(-0.04)
+	if Input.is_action_pressed("run"):
```

Comment:
Im assuming that this is responsible for the sprint function which is a good addition.